### PR TITLE
Add whitelist functionality

### PR DESCRIPTION
**Pull request by slhad**
As suggested by @jodem in https://github.com/jgsqware/clairctl/issues/65

and a little inspired by https://github.com/arminc/clair-scanner#example-whitelist-yaml-file

### CODE REVIEW

**Comment by ashmere** [comment]
+1 this would be a very useful feature for working with CI especially

**Comment by jgsqware** [comment]
is filtering priorities is not sufficient?
https://github.com/jgsqware/clairctl/blob/master/cmd/analyze.go#L155

**Comment by slhad** [comment]
@jgsqware filtering priorities is very usefull for rejecting a CI build but we still miss the capacity to filter out CVE which are actually not usable in some case.

For example, we use a layer from another people which provide a framework capable of using imagemagick, we know we will not use features using imagemagick but there is still some check at the startup of the framework we must use which require imgmagick, so we must leave the it but even if it's not relevent for us.

As long as we got the whitelist which contains explainations, it's good enougth for us, when reviewing it.

File: cmd/whitelist.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	Images           map[string]map[string]string
+}
+
+type WhiteList struct {
+	path        string
+	whitelisted vulnerabilitiesWhitelist
+}
+
+func (v *WhiteList) init() {
+	v.whitelisted = vulnerabilitiesWhitelist{}
```
Comment (jgsqware) [inline cmd/whitelist.go]: I would remove this and put this code diretly into NewWhiteList

File: cmd/whitelist.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	if err != nil {
+		log.Fatalf("unmarsh error %s : %v", v.path, err)
+	}
+}
+
+func (v *WhiteList) filter(analysis clair.ImageAnalysis) {
+
+	//access by ref to not reconstruct the whole struc
+
+	for indexLayerEnvelope := range analysis.Layers {
```
Comment (jgsqware) [inline cmd/whitelist.go]: Could you use hosrt name variable aka
```
indexLayerEnvelope -> i
IndexFeature -> f
```

it's a common pattern in go

File: cmd/whitelist.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+				}
+			}
+
+			analysis.Layers[indexLayerEnvelope].Layer.Features[indexFeature].Vulnerabilities = filteredVulnerabilities
+		}
+	}
+}
+
+func (v *WhiteList) isWhiteListed(namespace string, vulnerability string) bool {
+	imagesWhitelist := v.getImageWhiteList(namespace)
```
Comment (jgsqware) [inline cmd/whitelist.go]: you can avoid initialization using `v.getImageWhiteList(namespace)` in the if 

```go
if _, exists := v.getImageWhiteList(namespace)[vulnerability]; exists {
```

**Comment by slhad** [comment]
@jgsqware changes done

**Comment by jgsqware** [comment]
LGTM
Thanks you !

**Comment by slhad** [comment]
Thanks !

**Pull request merged**